src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,120 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.util.ConnectionUtil;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book save(Book book) {
+        String createQuery = "INSERT INTO book (title, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(createQuery,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows <= 0) {
+                throw new RuntimeException("Expected to insert 1 row or more");
+            }
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not save book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Book book;
+        String selectQuery = "SELECT * FROM book WHERE id=?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(selectQuery)) {
+            statement.setLong(1, id);
+
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                book = parseBook(resultSet);
+            } else {
+                throw new DataProcessingException("Expected not empty result.");
+            }
+            return Optional.of(book);
+        } catch (SQLException e) {
+            return Optional.empty();
+        }
```

Comment:
If SQLException occur then we can't say if we found something or not as we haven't proper response and we should cover such scenario into DataProcessingException.
On the other side if we haven't received data in result set but without exception then it means that all is fine but no result found. So we use Optional.ofNullable method to properly cover null book into empty optional
```suggestion
            ResultSet resultSet = statement.executeQuery();
            if (resultSet.next()) {
                book = parseBook(resultSet);
            }
            return Optional.ofNullable(book);
        } catch (SQLException e) {
            throw new DataProcessingException("Can not get book by id " + id, e);
        }
```